test/seqan3-test.cmake

```diff
@@ -48,22 +48,31 @@ include (FindPackageMessage)
 set (SEQAN3_BENCHMARK_CLONE_DIR "${PROJECT_BINARY_DIR}/vendor/benchmark")
 set (SEQAN3_TEST_CLONE_DIR "${PROJECT_BINARY_DIR}/vendor/googletest")
 
-# required flags, includes, definitions and libraries for seqan3
-set (SEQAN3_STRICT_CXX_FLAGS "-pedantic" "-Werror" "-Wall" "-Wextra")
-
-# required flags, includes and libraries for seqan3/test/performance
-set (SEQAN3_BENCHMARK_CXX_FLAGS ${SEQAN3_STRICT_CXX_FLAGS})
-set (SEQAN3_BENCHMARK_INCLUDE_DIRS "")
-set (SEQAN3_BENCHMARK_LIBRARIES "")
-list (APPEND SEQAN3_BENCHMARK_INCLUDE_DIRS "${SEQAN3_BENCHMARK_CLONE_DIR}/include/")
-list (APPEND SEQAN3_BENCHMARK_INCLUDE_DIRS "${SEQAN3_CLONE_DIR}/test/include/")
-
-# required flags, includes and libraries for seqan3/test/unit
-set (SEQAN3_TEST_CXX_FLAGS ${SEQAN3_STRICT_CXX_FLAGS})
-set (SEQAN3_TEST_INCLUDE_DIRS "")
-set (SEQAN3_TEST_LIBRARIES "")
-list (APPEND SEQAN3_TEST_INCLUDE_DIRS "${SEQAN3_TEST_CLONE_DIR}/googletest/include/")
-list (APPEND SEQAN3_TEST_INCLUDE_DIRS "${SEQAN3_CLONE_DIR}/test/include/")
+# seqan3::test exposes a base set of required flags, includes, definitions and
+# libraries qhich are in common for **all** seqan3 tests
+add_library (seqan3::test INTERFACE IMPORTED)
+set_property (TARGET seqan3::test APPEND PROPERTY INTERFACE_COMPILE_OPTIONS "-pedantic"  "-Wall" "-Wextra" "-Werror")
+set_property (TARGET seqan3::test APPEND PROPERTY INTERFACE_LINK_LIBRARIES "seqan3::seqan3" "pthread")
+set_property (TARGET seqan3::test APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES "${SEQAN3_CLONE_DIR}/test/include/")
```

Comment:
Here again, I am not sure, why these library targets need to be defined in the top level config cmake.
I'd like to have it as clean as possible in order to avoid the mess we had before. 
Is there any particular reason, of why these targets are exposed here and not within the project they are needed?
As far as I could see, there is no cross linking between other projects happening.

Reply:
This is the top-level-config for all test cases, not the top-level-config of `find(seqan3)`.

And `seqan3::test` is the parent (base class) to all kinds of tests (e.g. header test, unit test, performance test, coverage test), so it needs to be defined in the top-level.

There are also some library definition like `seqan3::test::unit`, which is used as a base for a couple of other kinds of test cases. For example `seqan3::test::coverage`, `seqan3::test::header` inherit from it. Thus, this needs to be defined in the top-level.

Reply:
>This is the top-level-config for all test cases, not the top-level-config of find(seqan3).

I know. But independent of this.

>And seqan3::test is the parent (base class) to all kinds of tests (e.g. header test, unit test, performance test, coverage test), so it needs to be defined in the top-level.
>There are also some library definition like seqan3::test::unit, which is used as a base for a couple of other kinds of test cases. For example seqan3::test::coverage, seqan3::test::header inherit from it. Thus, this needs to be defined in the top-level.

Ok, I just see now the dependencies. Alternatively, one could add a .cmake file into each subproject, but I see that the overhead might not be worthwhile. Could you add a section header above this definitions? Calling it `target link libraries for test projects` or something.